patches/server/0348-Add-ThrownEggHatchEvent.patch

```diff
@@ -7,18 +7,23 @@ Adds a new event similar to PlayerEggThrowEvent, but without the Player requirem
 (dispensers can throw eggs to hatch them, too).
 
 diff --git a/src/main/java/net/minecraft/world/entity/projectile/ThrownEgg.java b/src/main/java/net/minecraft/world/entity/projectile/ThrownEgg.java
-index 88e8ce770ddc61702fc87cd8dcee498183dc1032..4aca131fe774c54b5cc274000d825687a1373b3f 100644
+index 88e8ce770ddc61702fc87cd8dcee498183dc1032..97e6d07db5b0c5c4710605cc0fee8468583d109b 100644
 --- a/src/main/java/net/minecraft/world/entity/projectile/ThrownEgg.java
 +++ b/src/main/java/net/minecraft/world/entity/projectile/ThrownEgg.java
-@@ -82,6 +82,12 @@ public class ThrownEgg extends ThrowableItemProjectile {
+@@ -82,6 +82,17 @@ public class ThrownEgg extends ThrowableItemProjectile {
                      }
                  }
                  // CraftBukkit end
 +                // Paper start
 +                com.destroystokyo.paper.event.entity.ThrownEggHatchEvent event = new com.destroystokyo.paper.event.entity.ThrownEggHatchEvent((org.bukkit.entity.Egg) getBukkitEntity(), hatching, b0, hatchingType);
 +                event.callEvent();
 +                b0 = event.getNumHatches();
++                hatching = event.isHatching();
```

Comment:
or well, cleaner code would be
```java
hatching = event.isHatching();
b0 = hatching ? event.getNumHatches() : 0;
```

Reply:
oh github wasn't showing your last edit oO